File: internal/model/response/message.go
package response

import (
	"time"
)

type MessageResponse struct {
	ID             int                 `json:"id"`
	ConversationID int                 `json:"conversation_id"`
	SenderID       int                 `json:"sender_id"`
	Reply          *ReplyMetadata      `json:"reply"`
	Body           *string             `json:"body"`
	Deleted        bool                `json:"deleted"`
	CreatedAt      time.Time           `json:"created_at"`
	Attachments    []MessageAttachment `json:"attachments"`
}

type PaginatedMessageResponse struct {
	Messages       []MessageResponse `json:"messages"`
	NextCursor     *int              `json:"next_cursor"`
	PreviousCursor *int              `json:"previous_cursor"`
}

type ReplyMetadata struct {
	ID       int     `json:"id"`
	SenderID int     `json:"sender_id"`
	Body     *string `json:"body"`
	Deleted bool `json:"deleted"`
}

type MessageAttachment struct {
	ID       int    `json:"id"`
	Type     string `json:"type"`
	URL      string `json:"url"`
	Filename string `json:"filename"`
	Size     int64  `json:"size"`
}

File: frontend/src/types/http/message.ts
export type Message = {
    id: number
    conversation_id: number
    sender_id: number
    reply: ReplyMetadata | null
    body: string | null
    deleted: boolean
    created_at: string
    attachments: Attachment[]
}

export type ReplyMetadata = {
    id: number
    sender_id: number
    body: string | null
    deleted: boolean
}

export type Attachment = {
    id: number
    type: string
    url: string
    filename: string
    size: number
}

export type PaginatedMessage = {
    messages: Message[]
    next_cursor: number | null
    previous_cursor: number | null
}

export type MessageCreateRequest = {
    reply_to_id: number | null
    body: string | null
    attachments: AttachmentRequest[]
}

export type AttachmentRequest = {
    url: string
    filename: string
    type: string
    size: number
}

export type MessageEditRequest = {
    body: string
}